source/common/http/alternate_protocols_cache_impl.cc

```diff
@@ -28,15 +21,25 @@ AlternateProtocolsCacheImpl::findAlternatives(const Origin& origin) {
         nullptr);
   }
 
-  const Entry& entry = entry_it->second;
-  if (time_source_.monotonicTime() > entry.expiration_) {
-    // Expire the entry.
-    // TODO(RyanTheOptimist): expire entries based on a timer.
+  std::vector<AlternateProtocol>& protocols = entry_it->second;
+
+  const MonotonicTime now = time_source_.monotonicTime();
+  auto it = protocols.begin();
+  while (it != protocols.end()) {
```

Comment:
think it's worth writing this with std::remove_if?

Reply:
Done. It's slightly uglier than I would have expected because it needs a call to erase() as well as remove_if(). But it seems reasonable!